src/core/worker.lua

```diff
@@ -28,16 +28,23 @@ function start (name, luacode)
    if pid == 0 then
```

Comment:
That's a small variant of what I was working on when asked to PR this solution; I agree it's a much better one.